app/components/market/submit/submit-market-application.component.ts

```diff
@@ -1,208 +1,171 @@
             } else {
                 fg[key] = new FormControl();
             }
         }
-        this.jobFormGroup = this.formBuilder.group(fg);
-
-        this.form = this.formBuilder.group({
-            job: this.jobFormGroup,
-            pool: this.pickedPool,
-        });
-        if (this._formValue) {
-            this.form.setValue(this._formValue);
+        for (let key of poolParameters) {
+            if ("defaultValue" in this.poolTemplate.parameters[key]) {
+                const defaultValue = String(this.poolTemplate.parameters[key].defaultValue);
+                fg[key] = new FormControl(defaultValue);
+            } else {
+                fg[key] = new FormControl();
+            }
         }
-        this.form.valueChanges.debounceTime(400).distinctUntilChanged().subscribe((newFormValue) => {
-            this.router.navigate([], {
-                relativeTo: this.route,
-                queryParams: {
-                    formParams: JSON.stringify(newFormValue),
-                },
-            });
-        });
-    }
-
-    public getContainerFromFileGroup(fileGroup: string) {
-        return fileGroup && `fgrp-${fileGroup}`;
+        this.form = new FormGroup(fg);
     }
 
-    @autobind()
-    public submit() {
-        this.error = null;
-        const jobParams = this.jobFormGroup.value;
-        // RPC takes in Template JSON object and Parameter JSON object
-        const obs = this.pythonRpcService.callWithAuth("submit-ncj-job", [this._buildJobTemplate(), jobParams]);
+    private runJobWithPool(expandedPoolTemplate, jobInputs){
+        delete expandedPoolTemplate.id;
+        this.jobTemplate.job.properties.poolInfo = {
+            autoPoolSpecification: {
+            autoPoolIdPrefix: "jobname",
+            poolLifetimeOption: "job",
+            keepAlive: false,
+            pool: expandedPoolTemplate,
+            },
+        };
+        const obs = this.pythonRpcService.callWithAuth("submit-ncj-job", [this.jobTemplate, jobInputs]);
         obs.subscribe({
-            next: (data) => this._redirectToJob(data),
+            next: (data) => this._redirectToJob(),
             error: (err) => this.error = ServerError.fromPython(err),
         });
-        return obs;
     }
 
-    private _redirectToJob(data) {
-        const jobId = data.properties.id;
-        this.router.navigate(["/jobs", jobId]);
-    }
+    private onSubmit(){
+        this.error = null;
+        let jobInputs = {};
+        let poolInputs = {};
+        for (let param of Object.keys(this.form.controls)) {
+            if (this.getParameters(this.jobTemplate).includes(param)){
+                jobInputs[param] = this.form.controls[param].value;
+            }
+            else if (this.getParameters(this.poolTemplate).includes(param)) {
+                poolInputs[param] = this.form.controls[param].value;
+            }
+        }
+        if (this.mode_state === "PoolNJob"){
```

Comment:
"PoolNJob" should be changed to a const or enum. Probalby worth renaming to something more obvious.

Same comment goes for "PoolOJob" and "Pool"